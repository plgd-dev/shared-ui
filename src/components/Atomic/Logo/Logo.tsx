import { FC } from 'react'

import { Props } from './Logo.types'

const Logo: FC<Props> = (props) => {
    const { css, logo, className, onClick, title } = props
    return <img alt={title || ''} className={className} css={css} height={logo.height} onClick={onClick} src={logo.source} width={logo.width} />
}

Logo.displayName = 'Logo'

export default Logo
