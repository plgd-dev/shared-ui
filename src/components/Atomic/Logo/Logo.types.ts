export type Props = {
    css?: string
    logo: { height: string; width: string; source: string }
    className?: string
    onClick?: () => void
    title?: string
}
