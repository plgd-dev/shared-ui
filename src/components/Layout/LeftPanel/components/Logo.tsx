// Auto-generated file created by svgr-cli source svg-template.js
// Run yarn icons:create to update
// Do not edit
import * as React from 'react'
import { SVGProps } from 'react'
const SvgLogo = (props: SVGProps<SVGSVGElement>) => (
    <svg fill='none' height={props.height || 16} role='img' viewBox='0 0 147 32' width={props.width || 16} xmlns='http://www.w3.org/2000/svg' {...props}>
        <path
            clipRule='evenodd'
            d='M85.106 11.835c1.494 0 2.63.485 3.408 1.457v-1.305h2.836v8.483c0 1.867-.493 3.258-1.476 4.168-.984.907-2.42 1.362-4.308 1.362-.996 0-1.942-.12-2.834-.36-.895-.24-1.636-.588-2.222-1.042.119-.21 1.07-1.907 1.188-2.12.432.355.98.635 1.646.842.665.21 1.327.312 1.992.312 1.034 0 1.797-.229 2.288-.69.493-.462.738-1.16.738-2.095v-.434c-.779.847-1.866 1.27-3.256 1.27a5.53 5.53 0 0 1-2.596-.615 4.753 4.753 0 0 1-1.866-1.736c-.46-.742-.69-1.602-.69-2.574 0-.973.23-1.83.69-2.575a4.725 4.725 0 0 1 1.866-1.733c.786-.41 1.65-.615 2.596-.615Zm-19.016-.003c.958 0 1.83.218 2.617.655a4.71 4.71 0 0 1 1.845 1.835c.447.79.672 1.708.672 2.758 0 1.047-.225 1.965-.672 2.755a4.719 4.719 0 0 1-1.845 1.838c-.787.435-1.659.652-2.617.652-1.314 0-2.349-.41-3.102-1.233v4.756H60V11.983h2.854v1.175c.74-.883 1.82-1.325 3.236-1.325Zm63.871 0c.983 0 1.876.205 2.68.615a4.783 4.783 0 0 1 1.926 1.783c.478.775.718 1.695.718 2.755l-7.622 1.458c.217.505.559.884 1.024 1.137.465.253 1.037.378 1.714.378.536 0 1.011-.078 1.428-.235a3.48 3.48 0 0 0 1.158-.75l1.59 1.705c-.971 1.097-2.389 1.647-4.252 1.647-1.163 0-2.191-.222-3.084-.67-.895-.45-1.583-1.073-2.068-1.867-.485-.796-.728-1.698-.728-2.708 0-.998.24-1.897.718-2.7a5.004 5.004 0 0 1 1.972-1.875c.837-.448 1.779-.673 2.826-.673Zm-7.422-3.712v14.055h-2.854V21c-.741.885-1.816 1.325-3.219 1.325-.971 0-1.848-.215-2.634-.642a4.657 4.657 0 0 1-1.848-1.838c-.447-.795-.67-1.718-.67-2.765 0-1.047.225-1.97.67-2.765a4.658 4.658 0 0 1 1.848-1.837c.786-.43 1.663-.646 2.634-.646 1.315 0 2.344.413 3.085 1.233V8.12h2.988Zm-14.297 10.53c.524 0 .964.17 1.323.503.356.334.536.775.536 1.317 0 .53-.18.973-.536 1.326-.359.354-.799.53-1.323.53-.523 0-.965-.176-1.322-.53-.356-.354-.536-.796-.536-1.326 0-.542.18-.983.536-1.317.357-.333.799-.503 1.322-.503ZM76.202 8.12v10.512c0 .43.114.76.337.996.222.232.538.35.947.35.154 0 .309-.02.47-.058.16-.037.284-.083.372-.133.015.23.122 2.063.134 2.293a4.96 4.96 0 0 1-1.55.245c-1.162 0-2.07-.305-2.72-.917-.652-.613-.975-1.48-.975-2.605V8.12h2.985ZM104.471 8v14.055h-2.854V20.88c-.741.885-1.813 1.325-3.219 1.325-.968 0-1.848-.212-2.632-.642a4.653 4.653 0 0 1-1.848-1.838c-.447-.795-.67-1.718-.67-2.765 0-1.047.223-1.97.67-2.765a4.646 4.646 0 0 1 1.848-1.838c.784-.43 1.664-.642 2.632-.642 1.317 0 2.344.41 3.085 1.23V8h2.988Zm34.088 3.943 2.73 7.677 2.73-7.677H147l-4.098 10.225h-3.284l-4.098-10.225h3.039Zm-21.577 2.315c-.766 0-1.398.255-1.896.767-.498.51-.746 1.195-.746 2.055 0 .858.248 1.542.746 2.055.5.51 1.13.765 1.896.765.756 0 1.38-.255 1.878-.765.499-.513.746-1.197.746-2.055 0-.86-.247-1.545-.746-2.055-.498-.512-1.125-.767-1.878-.767Zm-51.408 0c-.766 0-1.395.255-1.886.767-.493.51-.738 1.195-.738 2.055 0 .858.245 1.542.738 2.055.49.51 1.12.765 1.886.765.767 0 1.396-.255 1.886-.765.49-.513.739-1.197.739-2.055 0-.86-.248-1.545-.74-2.055-.49-.512-1.118-.767-1.885-.767Zm33.343-.12c-.766 0-1.398.255-1.896.767-.498.51-.749 1.195-.749 2.055 0 .858.25 1.543.749 2.055.498.51 1.13.765 1.895.765.754.003 1.381-.255 1.876-.765.499-.512.749-1.197.749-2.055 0-.86-.25-1.545-.749-2.055-.498-.512-1.122-.768-1.876-.768Zm-13.217.12c-.791 0-1.443.23-1.954.692-.508.46-.763 1.063-.763 1.808 0 .744.255 1.347.763 1.81.51.46 1.163.69 1.954.69.791 0 1.438-.23 1.944-.69.506-.463.756-1.066.756-1.81 0-.745-.25-1.348-.756-1.808-.506-.463-1.153-.692-1.944-.692Zm44.261-.19c-.753 0-1.368.24-1.838.72-.473.48-.723 1.142-.748 1.99.501-.098 4.515-.87 5.018-.968a2.226 2.226 0 0 0-.862-1.268c-.435-.317-.958-.474-1.57-.474Z'
            fill='#191A1A'
            fillRule='evenodd'
        />
        <path
            clipRule='evenodd'
            d='M47.405 27.009A12.68 12.68 0 0 0 50 19.302c0-7.007-5.646-12.688-12.61-12.688-.566 0-1.122.042-1.669.115-.45-.723-.806-1.265-1.024-1.544C32.233 2.027 28.405 0 24.106 0c-6.37 0-11.704 4.456-13.098 10.438 0-.008-.008-.018-.006-.025-.1-.003-.199-.01-.3-.01C4.793 10.403 0 15.226 0 21.173 0 27.118 4.792 31.94 10.703 31.94c3.936 0 9.28.047 12.98.047 2.207 0 3.402-.026 12.513 0l.22-.002c1.184.056 3.729.011 6.147-1.109.388-.184 2.745-1.176 4.843-3.866l-.001-.001Z'
            fill='#FEBF40'
            fillRule='evenodd'
        />
        <path
            clipRule='evenodd'
            d='M47.405 27.009A12.68 12.68 0 0 0 50 19.302c0-7.007-5.646-12.688-12.61-12.688-.566 0-1.122.042-1.668.115-.45-.723-.806-1.265-1.025-1.544C32.233 2.027 28.406 0 24.106 0c-7.43 0-13.457 6.061-13.457 13.54 0 7.478 5.602 18.447 13.034 18.447 2.207 0 3.403-.026 12.513 0l.221-.002c1.183.056 3.728.011 6.147-1.109.388-.184 2.744-1.176 4.842-3.866l-.001-.001Z'
            fill='#87CEF2'
            fillRule='evenodd'
        />
        <path
            clipRule='evenodd'
            d='M46.93 26.296C44.882 22.622 36.278 7.21 34.696 5.184 32.233 2.027 28.404 0 24.105 0c-7.43 0-13.457 6.061-13.457 13.54 0 7.478 5.602 18.447 13.034 18.447 2.208 0 3.403-.026 12.513 0l.221-.002c1.184.056 3.729.012 6.147-1.109.388-.184 2.745-1.176 4.842-3.866l-.476-.714Z'
            fill='#2261AE'
            fillRule='evenodd'
        />
    </svg>
)
export default SvgLogo
