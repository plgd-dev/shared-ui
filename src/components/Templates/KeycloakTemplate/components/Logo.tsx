// Auto-generated file created by svgr-cli source svg-template.js
// Run yarn icons:create to update
// Do not edit
import * as React from 'react'
import { SVGProps } from 'react'
const SvgLogo = (props: SVGProps<SVGSVGElement>) => (
    <svg width={props.width || 16} height={props.height || 16} viewBox='0 0 140 32' fill='none' xmlns='http://www.w3.org/2000/svg' role='img' {...props}>
        <g clipPath='url(#logo_svg__a)' fillRule='evenodd' clipRule='evenodd' fill='#fff'>
            <path d='M78.05 17.237c1.502 0 2.642.505 3.424 1.52v-1.362h2.85v8.848c0 1.948-.496 3.398-1.484 4.347-.988.947-2.43 1.421-4.327 1.421-1 0-1.951-.125-2.847-.375-.9-.25-1.644-.613-2.233-1.087.12-.22 1.074-1.99 1.194-2.212.434.37.985.663 1.653.879a6.386 6.386 0 0 0 2.002.325c1.038 0 1.805-.239 2.298-.719.495-.482.742-1.21.742-2.185v-.454c-.783.884-1.875 1.325-3.272 1.325a5.39 5.39 0 0 1-2.608-.641 4.852 4.852 0 0 1-1.874-1.81c-.462-.775-.694-1.672-.694-2.686s.232-1.91.694-2.686a4.822 4.822 0 0 1 1.874-1.807 5.384 5.384 0 0 1 2.609-.641Zm-19.103-.003c.962 0 1.839.227 2.629.683a4.823 4.823 0 0 1 1.853 1.914c.45.824.676 1.781.676 2.876 0 1.092-.226 2.05-.676 2.874a4.834 4.834 0 0 1-1.853 1.917c-.79.453-1.666.68-2.63.68-1.32 0-2.358-.427-3.115-1.286v4.96h-3.002V17.39h2.867v1.226c.744-.92 1.829-1.382 3.25-1.382Zm64.165 0c.988 0 1.885.214 2.692.641a4.879 4.879 0 0 1 1.936 1.86c.48.808.721 1.768.721 2.873-.767.152-6.893 1.37-7.658 1.52.219.527.562.923 1.029 1.187.467.263 1.042.394 1.723.394.538 0 1.015-.081 1.434-.246a3.475 3.475 0 0 0 1.163-.782c.16.18 1.438 1.601 1.598 1.779-.975 1.144-2.4 1.718-4.272 1.718-1.168 0-2.202-.232-3.098-.699-.899-.469-1.59-1.119-2.078-1.948-.487-.829-.731-1.77-.731-2.824 0-1.04.241-1.979.721-2.816a5.113 5.113 0 0 1 1.981-1.956c.841-.466 1.788-.701 2.839-.701Zm-7.456-3.872v14.66h-2.868v-1.226c-.744.923-1.823 1.382-3.232 1.382-.976 0-1.857-.224-2.647-.67a4.775 4.775 0 0 1-1.857-1.917c-.449-.829-.673-1.792-.673-2.884 0-1.092.227-2.055.673-2.884a4.775 4.775 0 0 1 1.857-1.916c.79-.45 1.671-.673 2.647-.673 1.32 0 2.354.43 3.098 1.285v-5.157h3.002Zm-14.362 10.983c.525 0 .967.177 1.328.524.358.349.539.809.539 1.374 0 .553-.181 1.014-.539 1.383-.361.37-.803.552-1.328.552-.526 0-.971-.183-1.329-.552-.357-.369-.538-.83-.538-1.383 0-.565.18-1.025.538-1.374a1.841 1.841 0 0 1 1.329-.524ZM69.106 13.36v10.965c0 .45.114.794.338 1.038.223.243.54.366.952.366.155 0 .31-.021.472-.06.16-.04.285-.087.373-.139.016.24.122 2.152.135 2.391-.5.17-1.018.256-1.556.256-1.169 0-2.08-.318-2.734-.957-.655-.639-.98-1.543-.98-2.717V13.361h3Zm28.398-.125v14.66h-2.867v-1.225c-.744.923-1.82 1.382-3.233 1.382-.972 0-1.856-.222-2.644-.67a4.766 4.766 0 0 1-1.857-1.917c-.449-.829-.672-1.792-.672-2.884 0-1.092.223-2.055.672-2.884a4.76 4.76 0 0 1 1.857-1.917c.788-.448 1.672-.67 2.644-.67 1.323 0 2.355.428 3.099 1.283v-5.158h3.001Zm34.246 4.113 2.743 8.008 2.742-8.008h2.995l-4.117 10.665h-3.299l-4.117-10.665h3.053Zm-21.677 2.414c-.769 0-1.404.266-1.904.8-.501.533-.75 1.247-.75 2.144 0 .895.249 1.609.75 2.144.503.532 1.135.798 1.904.798.76 0 1.387-.266 1.887-.798.501-.535.75-1.25.75-2.144 0-.897-.249-1.611-.75-2.143-.5-.535-1.13-.8-1.887-.8Zm-51.644 0c-.77 0-1.402.266-1.895.8-.495.533-.741 1.247-.741 2.144 0 .895.246 1.609.741 2.144.493.532 1.125.798 1.895.798s1.402-.266 1.894-.798c.493-.535.742-1.25.742-2.144 0-.897-.249-1.611-.742-2.143-.492-.535-1.124-.8-1.894-.8Zm33.496-.125c-.77 0-1.404.266-1.905.8-.5.533-.752 1.247-.752 2.144 0 .895.252 1.609.752 2.144.5.532 1.135.797 1.905.797.757.003 1.386-.265 1.884-.797.5-.535.752-1.25.752-2.144 0-.897-.251-1.611-.752-2.144-.5-.534-1.127-.8-1.884-.8Zm-13.278.125c-.795 0-1.45.24-1.963.722-.51.48-.767 1.109-.767 1.886 0 .777.257 1.405.767 1.888.513.48 1.168.72 1.963.72.795 0 1.445-.24 1.953-.72.508-.483.76-1.111.76-1.888 0-.777-.252-1.406-.76-1.886-.508-.482-1.158-.722-1.953-.722Zm44.465-.198c-.757 0-1.374.25-1.846.75-.475.501-.726 1.193-.752 2.077.503-.102 4.536-.908 5.042-1.01a2.329 2.329 0 0 0-.867-1.322c-.436-.33-.962-.495-1.577-.495Z' />
            <path
                d='M46.424 26.992a12.84 12.84 0 0 0 2.542-7.702c0-7.002-5.53-12.68-12.35-12.68-.554 0-1.099.041-1.633.115-.442-.723-.79-1.265-1.004-1.544C31.566 2.026 27.818 0 23.607 0 17.37 0 12.145 4.453 10.78 10.431c0-.008-.007-.017-.005-.025-.098-.003-.195-.009-.294-.009C4.693 10.397 0 15.216 0 21.159 0 27.102 4.693 31.92 10.481 31.92c3.855 0 9.088.047 12.712.047 2.161 0 3.332-.026 12.254 0l.216-.002c1.16.055 3.651.011 6.02-1.108.38-.184 2.688-1.175 4.742-3.864v-.001Z'
                fillOpacity={0.3}
            />
            <path
                d='M46.424 26.992a12.84 12.84 0 0 0 2.542-7.702c0-7.002-5.53-12.68-12.35-12.68-.554 0-1.099.041-1.633.115-.442-.723-.79-1.265-1.004-1.544C31.566 2.026 27.818 0 23.607 0 16.331 0 10.43 6.058 10.43 13.53c0 7.475 5.486 18.437 12.764 18.437 2.161 0 3.332-.026 12.254 0l.216-.002c1.16.055 3.651.011 6.02-1.108.38-.184 2.688-1.175 4.742-3.864v-.001Z'
                fillOpacity={0.6}
            />
            <path d='M45.959 26.28c-2.004-3.672-10.43-19.075-11.98-21.099-2.412-3.155-6.161-5.18-10.372-5.18-7.276 0-13.178 6.057-13.178 13.53 0 7.474 5.485 18.436 12.764 18.436 2.162 0 3.332-.025 12.254 0l.216-.002c1.16.056 3.652.012 6.02-1.108.38-.184 2.688-1.175 4.742-3.864l-.466-.713Z' />
        </g>
        <defs>
            <clipPath id='logo_svg__a'>
                <path fill='#fff' d='M0 0h140v32H0z' />
            </clipPath>
        </defs>
    </svg>
)
export default SvgLogo
