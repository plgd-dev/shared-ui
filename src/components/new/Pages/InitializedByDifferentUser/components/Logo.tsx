// Auto-generated file created by svgr-cli source svg-template.js
// Run yarn icons:create to update
// Do not edit
import * as React from 'react'
import { SVGProps } from 'react'
const SvgLogo = (props: SVGProps<SVGSVGElement>) => (
    <svg width={props.width || 16} height={props.height || 16} viewBox='0 0 170 38' fill='none' xmlns='http://www.w3.org/2000/svg' role='img' {...props}>
        <path
            fillRule='evenodd'
            clipRule='evenodd'
            d='M100.28 15.261c1.683 0 2.962.539 3.839 1.62v-1.45h3.195v9.425c0 2.074-.555 3.62-1.663 4.63-1.108 1.009-2.725 1.514-4.853 1.514-1.121 0-2.187-.133-3.192-.4-1.008-.267-1.843-.653-2.503-1.158.134-.233 1.205-2.12 1.339-2.356.486.395 1.104.706 1.853.936.75.234 1.495.347 2.244.347 1.165 0 2.025-.255 2.577-.766.556-.514.832-1.29.832-2.328v-.483c-.877.942-2.102 1.41-3.668 1.41-1.065 0-2.039-.227-2.924-.682a5.326 5.326 0 0 1-2.102-1.928c-.518-.825-.777-1.78-.777-2.861 0-1.08.259-2.034.777-2.862a5.295 5.295 0 0 1 2.102-1.925c.885-.455 1.86-.683 2.924-.683Zm-21.42-.003c1.08 0 2.062.242 2.948.728a5.27 5.27 0 0 1 2.078 2.039c.504.878.758 1.898.758 3.064 0 1.163-.254 2.183-.758 3.061a5.28 5.28 0 0 1-2.078 2.042c-.886.483-1.869.725-2.948.725-1.48 0-2.645-.456-3.494-1.37v5.284H72V15.425h3.215v1.306c.834-.98 2.05-1.473 3.645-1.473Zm71.947 0c1.107 0 2.113.228 3.018.684a5.356 5.356 0 0 1 2.17 1.98c.538.86.809 1.884.809 3.061l-8.586 1.62c.245.561.629.983 1.154 1.264.523.28 1.167.42 1.931.42.603 0 1.138-.087 1.608-.262.467-.177.9-.452 1.304-.833l1.791 1.895c-1.093 1.219-2.691 1.83-4.789 1.83-1.31 0-2.469-.247-3.474-.745-1.008-.5-1.783-1.192-2.33-2.075-.547-.883-.82-1.886-.82-3.008 0-1.109.271-2.108.809-3a5.6 5.6 0 0 1 2.221-2.083c.943-.498 2.004-.748 3.184-.748Zm-8.361-4.124V26.75h-3.215v-1.305c-.834.983-2.045 1.472-3.625 1.472-1.094 0-2.082-.24-2.968-.714a5.214 5.214 0 0 1-2.081-2.042c-.504-.883-.755-1.909-.755-3.072 0-1.164.254-2.19.755-3.072a5.214 5.214 0 0 1 2.081-2.042c.886-.478 1.874-.717 2.968-.717 1.48 0 2.64.459 3.474 1.37v-5.494h3.366Zm-16.104 11.7c.59 0 1.085.188 1.489.558.402.372.604.861.604 1.463 0 .59-.202 1.081-.604 1.473-.404.394-.899.589-1.489.589-.589 0-1.088-.195-1.489-.589-.401-.392-.604-.883-.604-1.473 0-.602.203-1.091.604-1.463.401-.37.9-.558 1.489-.558Zm-36.091-11.7v11.68c0 .478.128.845.378 1.105.251.26.607.39 1.068.39.174 0 .347-.023.53-.064.18-.042.319-.092.418-.148.017.256.137 2.292.152 2.547a5.654 5.654 0 0 1-1.746.273c-1.31 0-2.332-.34-3.065-1.02-.734-.68-1.098-1.644-1.098-2.894v-11.87h3.363ZM122.093 11v15.617h-3.215V25.31c-.834.983-2.041 1.472-3.625 1.472-1.09 0-2.081-.236-2.964-.713a5.204 5.204 0 0 1-2.082-2.042c-.504-.883-.755-1.909-.755-3.073 0-1.163.251-2.188.755-3.072a5.197 5.197 0 0 1 2.082-2.041c.883-.478 1.874-.715 2.964-.715 1.484 0 2.64.456 3.474 1.368V11h3.366Zm38.398 4.381 3.076 8.53 3.075-8.53H170l-4.616 11.361h-3.699l-4.616-11.361h3.422Zm-24.304 2.572c-.863 0-1.575.284-2.136.853-.561.566-.84 1.327-.84 2.283 0 .953.279 1.714.84 2.284.564.566 1.273.85 2.136.85.851 0 1.554-.284 2.115-.85.562-.57.841-1.331.841-2.284 0-.956-.279-1.717-.841-2.283-.561-.57-1.267-.853-2.115-.853Zm-57.908 0c-.863 0-1.572.284-2.124.853-.556.566-.832 1.327-.832 2.283 0 .953.276 1.714.832 2.284.552.566 1.261.85 2.124.85.863 0 1.572-.284 2.124-.85.553-.57.832-1.331.832-2.284 0-.956-.279-1.717-.832-2.283-.552-.57-1.26-.853-2.124-.853Zm37.558-.134c-.863 0-1.574.284-2.136.853-.56.567-.843 1.328-.843 2.283 0 .954.283 1.715.843 2.284.562.567 1.273.85 2.136.85.849.003 1.555-.283 2.113-.85.561-.57.843-1.33.843-2.284 0-.955-.282-1.716-.843-2.283-.561-.57-1.264-.853-2.113-.853Zm-14.888.134c-.891 0-1.626.255-2.2.77-.573.51-.86 1.18-.86 2.008 0 .827.287 1.497.86 2.011.575.51 1.309.767 2.2.767.892 0 1.621-.256 2.19-.767.57-.514.852-1.184.852-2.011 0-.828-.282-1.498-.852-2.009-.569-.514-1.298-.77-2.19-.77Zm49.858-.211c-.849 0-1.541.267-2.07.8-.533.533-.815 1.27-.844 2.211l5.653-1.075a2.472 2.472 0 0 0-.971-1.409c-.49-.352-1.079-.527-1.768-.527Z'
            fill='#191A1A'
        />
        <path
            fillRule='evenodd'
            clipRule='evenodd'
            d='M52.77 32.541a13.71 13.71 0 0 0 2.889-8.429c0-7.663-6.285-13.876-14.037-13.876-.63 0-1.25.045-1.858.125-.5-.791-.898-1.383-1.14-1.689a15.015 15.015 0 0 0-11.79-5.67c-7.09 0-13.028 4.873-14.58 11.416 0-.009-.009-.02-.007-.028-.111-.003-.221-.01-.333-.01C5.334 14.38 0 19.654 0 26.158c0 6.503 5.335 11.777 11.914 11.777 4.382 0 10.33.05 14.45.05 2.456 0 3.787-.027 13.928 0 .084.001.163-.001.246-.001 1.318.06 4.15.012 6.843-1.213.431-.202 3.055-1.286 5.39-4.228l-.001-.002Z'
            fill='#FEBF40'
        />
        <path
            fillRule='evenodd'
            clipRule='evenodd'
            d='M57.11 29.54A13.71 13.71 0 0 0 60 21.11c0-7.663-6.285-13.876-14.037-13.876-.63 0-1.25.045-1.857.125-.501-.791-.898-1.383-1.141-1.689A15.015 15.015 0 0 0 31.175 0c-8.271 0-14.98 6.63-14.98 14.808 0 8.179 6.236 20.176 14.51 20.176 2.456 0 3.787-.028 13.929 0l.245-.002c1.318.06 4.15.012 6.843-1.213.431-.201 3.055-1.286 5.39-4.228l-.001-.002Z'
            fill='#87CEF2'
        />
        <path
            fillRule='evenodd'
            clipRule='evenodd'
            d='M52.24 31.762c-2.277-4.019-11.855-20.875-13.616-23.09a15.015 15.015 0 0 0-11.79-5.67c-8.271 0-14.98 6.629-14.98 14.807 0 8.18 6.236 20.176 14.509 20.176 2.457 0 3.788-.027 13.93 0l.245-.002c1.317.061 4.15.013 6.842-1.212.433-.202 3.056-1.287 5.39-4.229l-.53-.78Z'
            fill='#2261AE'
        />
    </svg>
)
export default SvgLogo
