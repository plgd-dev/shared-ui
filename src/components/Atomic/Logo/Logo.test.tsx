import { render, fireEvent } from '@testing-library/react'
import Logo from './Logo'

describe('Logo', () => {
    it('renders without crashing', () => {
        const { container, asFragment } = render(<Logo logo={{ height: '50px', width: '50px', source: 'logo.png' }} />)
        expect(container).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders with correct source', () => {
        const { getByAltText } = render(<Logo logo={{ height: '50px', width: '50px', source: 'logo.png' }} />)
        expect(getByAltText('')).toHaveAttribute('src', 'logo.png')
    })

    it('renders with correct dimensions', () => {
        const { getByAltText } = render(<Logo logo={{ height: '50px', width: '50px', source: 'logo.png' }} />)
        const logo = getByAltText('')
        expect(logo).toHaveAttribute('height', '50px')
        expect(logo).toHaveAttribute('width', '50px')
    })

    it('calls onClick when clicked', () => {
        const onClick = jest.fn()
        const { getByAltText } = render(<Logo logo={{ height: '50px', width: '50px', source: 'logo.png' }} onClick={onClick} />)
        fireEvent.click(getByAltText(''))
        expect(onClick).toHaveBeenCalled()
    })

    it('does not call onResized when collapsed', () => {
        const onResized = jest.fn()
        const { getByAltText } = render(<Logo collapsed={true} logo={{ height: '50px', width: '50px', source: 'logo.png' }} onResized={onResized} />)
        fireEvent.click(getByAltText(''))
        expect(onResized).not.toHaveBeenCalled()
    })
})
