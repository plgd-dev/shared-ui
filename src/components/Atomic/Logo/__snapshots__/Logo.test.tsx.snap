// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders without crashing 1`] = `
<DocumentFragment>
  <img
    alt=""
    class="emotion-0"
    height="50px"
    id="main-app-logo"
    src="logo.png"
    width="50px"
  />
</DocumentFragment>
`;
